File: my-solutions/0102-binary-tree-level-order-traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def traverseTree(self, root, level, level_map):
        if root == None:
            return level_map
        if level in level_map.keys():
            level_map[level].append(root.val)
        else:
            level_map[level] = [root.val]
        self.traverseTree(root.left, level + 1, level_map)
        self.traverseTree(root.right, level + 1, level_map)
        return level_map
        
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        # traversal
        level_map = self.traverseTree(root, 1, {})
        return list(level_map.values())
        
        

File: my-solutions/0102-binary-tree-level-order-traversal/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
#include <queue>

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(root == nullptr) {
            return vector<vector<int>>();
        }
        vector<vector<int>> result;
        vector<int> level;
        queue<TreeNode*> q;
        q.push(root);
        int count = 1;
        while(!q.empty()) {
            level.clear();
            for(int i = 0; i < count; i++) {
                if(q.front()) {
                    level.push_back(q.front()->val);
                    if(q.front()->left) q.push(q.front()->left);
                    if(q.front()->right) q.push(q.front()->right);
                    q.pop();
                }
            }
            count = size(q);
            result.push_back(level);
        }
        return result;
    }
};
